server/src/main/java/org/allaymc/server/registry/loader/PackRegistryLoader.java

```diff
@@ -124,12 +125,19 @@ private void encryptPacks() {
                 String key;
                 try {
                     Files.move(zipPack, tmpPath);
-                    key = PackEncryptor.encrypt(tmpPath, zipPack);
+                    if (isKeyPathExists){
+                        key = Files.readString(keyPath);
+                        PackEncryptor.encrypt(tmpPath, zipPack, key);
```

Comment:
I think this will make a pack being encrypted multiple times, which is an unexpected behavior

Reply:
But I have check for .bak file in the beginning

Reply:
Maybe check the content.json thing

Reply:
yeah that make sense i gonna close this pr